packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/parameters.ts

```diff
@@ -0,0 +1,142 @@
+// Licensed under the MIT license.
+
+import { RequestParameters } from "@typespec/ts-http-runtime";
+import { User, TodoItem, TodoUrlAttachment, TodoItemPatch } from "./models";
+
+export interface UsersCreateBodyParam {
+  body?: { user: User };
+}
+
+export type UsersCreateParameters = UsersCreateBodyParam & RequestParameters;
+
+export interface UsersValidateQueryParamProperties {
+  token: string;
+}
+
+export interface UsersValidateQueryParam {
+  queryParameters: UsersValidateQueryParamProperties;
+}
+
+export type UsersValidateParameters = UsersValidateQueryParam &
+  RequestParameters;
+
+export interface UsersLoginBodyParam {
+  body?: { username: string; password: string };
+}
+
+export type UsersLoginParameters = UsersLoginBodyParam & RequestParameters;
+export type UsersLogoutParameters = RequestParameters;
+
+export interface UsersForgotPasswordBodyParam {
+  body?: { email: string };
+}
+
+export type UsersForgotPasswordParameters = UsersForgotPasswordBodyParam &
+  RequestParameters;
+
+export interface UsersResetPasswordQueryParamProperties {
+  resetToken: string;
+}
+
+export interface UsersResetPasswordQueryParam {
+  queryParameters: UsersResetPasswordQueryParamProperties;
+}
+
+export type UsersResetPasswordParameters = UsersResetPasswordQueryParam &
+  RequestParameters;
+
+export interface TodoItemsListQueryParamProperties {
+  /** The limit to the number of items */
+  limit: number;
+  /** The offset to start paginating at */
+  offset: number;
+}
+
+export interface TodoItemsListQueryParam {
+  queryParameters: TodoItemsListQueryParamProperties;
+}
+
+export type TodoItemsListParameters = TodoItemsListQueryParam &
+  RequestParameters;
+
+export interface TodoItemsCreateJsonBodyParam {
+  body?: { item: TodoItem; attachments: Array<TodoUrlAttachment> };
+}
+
+export interface TodoItemsCreateJsonMediaTypesParam {
+  contentType: "application/json";
+}
+
+export type TodoItemsCreateJsonParameters = TodoItemsCreateJsonMediaTypesParam &
+  TodoItemsCreateJsonBodyParam &
+  RequestParameters;
+
+export interface TodoItemsCreateFormBodyParam {
+  body?: {
+    item: TodoItem;
+    attachments?: (
+      | TodoUrlAttachment
+      | string
+      | Uint8Array
```

Comment:
With the [fix](https://github.com/Azure/autorest.typescript/pull/2258) we could generate the union of following for MFD. And @timovv would work on the sync between branding and non-branding.

@joheredi Do you think we could merge the codegen pr?
https://github.com/Azure/autorest.typescript/pull/2300/commits/d10ecbc75a1c2ff7529b16a3917796ca23f0ef25